src/matlab2tikz.m

```diff
@@ -4005,22 +4010,54 @@ function warnFacetedInterp(m2t)
     else
         arrowLength = '{sqrt((\thisrow{u})^2+(\thisrow{v})^2)}';
     end
+
     plotOpts = opts_add(plotOpts, 'point meta', arrowLength);
+    plotOpts = opts_add(plotOpts, 'point meta min', '0');
 
     if showArrowHead
-        %TODO: scale the arrows more rigorously to match MATLAB behavior
-        %There is a "MaxHeadSize" property (at least in R2014b) that plays a
-        %role in determining the quiver size.
         arrowHeadOpts = opts_new();
-        arrowHeadOpts = opts_add(arrowHeadOpts, 'scale length', ...
-                                 '{max(0.01,\pgfplotspointmetatransformed/1000)}');
-        arrowHeadOpts = opts_add(arrowHeadOpts, 'scale width', ...
-                                 '{0.5*max(0.01,\pgfplotspointmetatransformed/1000)}');
+
+        % In MATLAB (HG1), the arrow head is constructed to have an angle of
+        % approximately 18.263 degrees in 2D as can be derived from the
+        % |quiver| function.
+        % In 3D, the angle is no longer constant but it is approximately
+        % the same as for 2D quiver plots. So let's make our life easy.
+        % |test/examples/example_quivers.m| covers the calculations.
+        arrowHeadOpts = opts_add(arrowHeadOpts, 'angle''', '18.263');
+
+        % TODO: scale the arrows more rigorously to match MATLAB behavior
+        % Currently, this is quite hard to do, since the size of the arrows
+        % is defined in pgfplots in absolute units (here we specify that those
+        % should be scaled up/down according to the data), while the data itself
+        % is in axis coordinates (or some scaled variant). I.e. we need the
+        % physical dimensions of the axis to compute the correct scaling!
+        % 
+        % There is a "MaxHeadSize" property that plays a role.
+        % MaxHeadSize is said to be relative to the length of the quiver in the
+        % MATLAB documentation. However, in practice, there seems to be a SQRT
+        % involved somewhere (e.g. if u.^2 + v.^2 == 2, all MHS values >
+        % 1/sqrt(2) are capped to 1/sqrt(2)).
+        %
+        % NOTE: `set(h, 'MaxHeadSize')` is bugged in HG1 (not in HG2 or Octave)
+        % according to http://www.mathworks.com/matlabcentral/answers/96754
+
+        userInfo(m2t, ['Please change the "arrowHeadSize" option', ...
+                       ' if the size of the arrows is incorrect.']);
+        arrowHeadSize = sprintf(m2t.ff, abs(m2t.cmdOpts.Results.arrowHeadSize));
+
+        % Write out the actual scaling for TikZ.
+        % `\pgfplotspointsmetatransformed` is in the range [0, 1000], so
+        % divide by this span (as is done in the pgfplots manual) to normalize
+        % the arrow head size.
+        arrowHeadOpts = opts_add(arrowHeadOpts, 'scale', ...
+              ['{' arrowHeadSize '*\pgfplotspointmetatransformed/1000}']);
```

Comment:
So out of curriosity. If \pgfplotspointmetatransformed ranges from [0, 1000] and we scale it to [0, 1]. Wouldnt that mean, that a arrowHeadSize ~= 10 / 1000 = 0.01 which is 20 times smaller than the 0.2 in matlab?

Could you check the difference when setting this to 200?